lib/auth/managed_authenticator.rb

```diff
@@ -140,7 +140,7 @@ def find_user_by_email(auth_token)
   def find_user_by_username(auth_token)
     username = auth_token.dig(:info, :nickname)
     if username
-      User.find_by_username(username)
+      User.find_by_username_lower(username.downcase)
     end
```

Comment:
`find_by_username` is the correct thing to do here - that will handle case-insentivity (and some other normalization) already:

https://github.com/discourse/discourse/blob/dc89b72d465329b041b0b26be4e6176343288176/app/models/user.rb#L411-L413

Reply:
Ah, I was being too careful!